### ISSUE TITLE
Improper handling of CORS preflight checks

### ISSUE THREAD

**Issue by individuwill** [open-meteo/open-meteo#526]
I have written a small locally hosted SPA using htmx that connects to the public open-meteo weather service forecast API endpoint. I'm experiencing a CORS related error when viewing the SPA in Firefox.

The issue is that the htmx library uses the browser's XMLHTTPRequest API to hit the open-meteo API endpoint and this library also attaches event listeners to the `xhr.upload` property even when not uploading. The library does not offer a way to disable this functionality. In Firefox (and perhaps webkit) simply attaching an event listener to this property triggers a CORS preflight check. The open-meteo public API does not properly handle a CORS preflight check.

The CORS preflight consists of sending an HTTP OPTIONS method to the desired endpoint and examining the response headers to determine if the request can move forward. Unfortunately the open-meteo API endpoint does not currently respond properly to the HTTP OPTIONS method and so the CORS preflight check fails.

There are a bunch of different ways to fix this, but it seems that the intent for the open-meteo API is to support CORS as the proper headers are configured and returned for a carefully constructed GET request (just not the required OPTIONS request). So I think the fix can / should be applied to the open-meteo API.

Here is a minimal example of the SPA app. This HTML can be placed in a file and opened in the browser (Firefox is the one that has issues with this). If you open the developer tools, you'll see the CORS errors in the console and in the network pane.

```html
<!DOCTYPE html>
<html lang="en">

<head>
    <script src="https://unpkg.com/htmx.org@1.9.9"></script>
    <script>
        document.addEventListener('htmx:configRequest', (evt) => {
            // This fix is needed to satisfy some CORS conditions
            evt.detail.headers = [];
        });
    </script>
</head>

<body>
    <div hx-get="http://api.open-meteo.com/v1/forecast?latitude=37.0902&longitude=95.7129&daily=weather_code,temperature_2m_max,temperature_2m_min,uv_index_max,precipitation_probability_max,wind_speed_10m_max,wind_gusts_10m_max&temperature_unit=fahrenheit&wind_speed_unit=mph&precipitation_unit=inch&timezone=America%2FLos_Angeles"
        hx-swap="innerText" hx-trigger="load">
    </div>
</body>

</html>
```

To simlate the issue on the command line you can use a `curl` command. Below are 2 samples of this, one with the GET method, the other with the OPTIONS method.

Minimal GET public:

```bash
curl -X GET -D - \
    'https://api.open-meteo.com/v1/forecast?latitude=37.0902&longitude=95.7129&daily=weather_code,temperature_2m_max,temperature_2m_min,uv_index_max,precipitation_probability_max,wind_speed_10m_max,wind_gusts_10m_max&temperature_unit=fahrenheit&wind_speed_unit=mph&precipitation_unit=inch&timezone=America%2FLos_Angeles' \
    -H 'Origin: null'

HTTP/2 200 
date: Mon, 27 Nov 2023 19:59:47 GMT
content-type: application/json; charset=utf-8
access-control-allow-origin: *
access-control-allow-headers: accept, authorization, content-type, origin, x-requested-with
access-control-allow-methods: GET, OPTIONS
access-control-max-age: 600

{"latitude":37.0,"longitude":95.875 ...
```

Minimal OPTIONS public:

```bash
curl -X OPTIONS -D - \
    'https://api.open-meteo.com/v1/forecast?latitude=37.0902&longitude=95.7129&daily=weather_code,temperature_2m_max,temperature_2m_min,uv_index_max,precipitation_probability_max,wind_speed_10m_max,wind_gusts_10m_max&temperature_unit=fahrenheit&wind_speed_unit=mph&precipitation_unit=inch&timezone=America%2FLos_Angeles' \
    -H 'Origin: null'

HTTP/2 404 
date: Mon, 27 Nov 2023 20:01:20 GMT
content-type: application/json; charset=utf-8
content-length: 35

{"error":true,"reason":"Not Found"}
```

**Comment by patrick-zippenfenig** [comment]
Thanks! Ideally CORS would be handled correctly by the [Vapor CORS middleware](https://docs.vapor.codes/advanced/middleware/?h=cors#cors-middleware). I am not sure if adding `options` as a valid method could have side effects. E.g. effectively running the same API call twice. I will go through the Vapor code to check how it is handled

**Comment by patrick-zippenfenig** [comment]
I applied a fix in #528. However, it will take a day until the change is live on public open-meteo servers
